File: OOps/Encapsulation.ts
class BankAccount {
  // Using 'private' enforces true encapsulation, unlike Python's underscore convention.
  // These properties cannot be accessed directly from outside the class.
  private accountNumber: string;
  private balance: number;

  constructor(accountNumber: string, balance: number) {
    this.accountNumber = accountNumber;
    this.balance = balance;
  }

  deposit(amount: number): void {
    this.balance += amount;
  }

  withdraw(amount: number): void {
    if (this.balance >= amount) {
      this.balance -= amount;
    } else {
      console.log("Insufficient funds");
    }
  }

  getBalance(): number {
    return this.balance;
  }
}

File: OOps/Encapsulation.py
class BankAccount:
    def __init__(self, account_number, balance):
        # Encapsulation example with private variables
        self._account_number = account_number
        self._balance = balance
    
    def deposit(self, amount):
        self._balance += amount
        
    def withdraw(self, amount):
        if self._balance >= amount:
            self._balance -= amount
        else:
            print("Insufficient funds")
            
    def get_balance(self):
        return self._balance